pkg/operation/botanist/logging.go

```diff
@@ -70,3 +91,20 @@ func (b *Botanist) DeploySeedLogging(ctx context.Context) error {
 
 	return b.K8sSeedClient.ChartApplier().Apply(ctx, filepath.Join(charts.Path, "seed-bootstrap", "charts", "loki"), b.Shoot.SeedNamespace, fmt.Sprintf("%s-logging", b.Shoot.SeedNamespace), kubernetes.Values(lokiValues))
 }
+
+func (b *Botanist) isShootNodeLoggingActivated() bool {
+	if gardenletfeatures.FeatureGate.Enabled(features.Logging) && b.Shoot != nil &&
+		b.Shoot.Purpose != gardencorev1beta1.ShootPurposeTesting && b.Config != nil &&
+		b.Config.Logging != nil && b.Config.Logging.ShootNodeLogging != nil {
+
+		return b.Config.Logging.ShootNodeLogging.AllowAllExecptTesting || func(allowedPurposes []string, purpose string) bool {
+			for _, allowedallowedPurpose := range allowedPurposes {
```

Comment:
Can you find better variable names?

Reply:
Hmm, This was mistake, maybe some sort of autocompletion suggestion after typing a`llowed` once.